extension/ts/extensions/author/authorPage/updateBook.ts

```diff
@@ -0,0 +1,71 @@
+import {AuthorRecord} from "../../../adapters/author";
+import {BookRecord} from "../../../adapters/book/index";
+import {filterAuthorTags} from "../../../util/filterAuthorTags";
+
+const getOtherAuthorTags = async (
+	book: BookRecord,
+	currentAuthor: string,
+	getAuthor: (uuid: string) => Promise<AuthorRecord>
+) => {
+	const futureOtherAuthors = book.authorIds.filter((author) => author !== currentAuthor).map(getAuthor);
+	const otherAuthors = await Promise.all(futureOtherAuthors);
+	return filterAuthorTags(otherAuthors.flatMap((author) => author?.tags ?? []));
+};
+
+const getBookOnlyTags = (tags: string[]) => {
+	const authorTags = filterAuthorTags(tags);
+	return tags.filter((tag) => !authorTags.includes(tag));
+};
+
+/**
+ * yes, i know this code is unreadable, hence the comments
+ *
+ * perhaps one day i will come in and clean it up -- i think it could be simpler
+ */
+const createUpdateBook =
+	(getAuthor: (currentAuthorId: string) => Promise<AuthorRecord>, saveBook: (book: BookRecord) => Promise<void>) =>
+	(uuid: string, tags: string[]) =>
+	async (book: BookRecord): Promise<BookRecord> => {
+		try {
+			// get only the author tags for current book
+			const bookTags = filterAuthorTags(book.tags);
+			// might need to delete author tags not on this author
+			const maybeDeleteTags = bookTags.filter((tag) => !tags.includes(tag));
+			// first get all authors that aren't the current one
+			const otherAuthorTags = await getOtherAuthorTags(book, uuid, getAuthor);
+			// if a tag isn't on any author, it should be deleted
+			const deleteTags = maybeDeleteTags.filter((tag) => !otherAuthorTags.includes(tag));
+			// finally, all tags except the deleted ones
+			const allTags = [...new Set([...book.tags, ...filterAuthorTags(tags)])];
+			book.tags = allTags.filter((tag) => !deleteTags.includes(tag));
+			await saveBook(book);
+			return book;
+		} catch (error) {
+			console.error(book, error);
+			return null;
+		}
+	};
+
+/**
+ * This version syncs every book with all of its authors! And it's easier to read!
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const alternate =
```

Comment:
@adirar111 I think I might want to change to this version!

- \+ It's easier to read
- \+ It actually does more, as it syncs the book with all its authors
- \- I worry that it's more difficult for a user to reason about, as a book might start getting tags added to it from another co-author that the user isn't on the page for editing

Reply:
OK made the change